packages/common/base/src/apiClient/ApiClient.test.ts

```diff
@@ -0,0 +1,135 @@
+import { describe, expect, test, vi, beforeEach } from "vitest";
+import { ApiClient } from "./ApiClient";
+
+// Mock fetch globally
+const mockFetch = vi.fn();
+global.fetch = mockFetch;
+
+class TestApiClient extends ApiClient {
+    get authHeaders(): HeadersInit {
+        return { "x-api-key": "test-key" };
+    }
+
+    get baseUrl(): string {
+        return "https://api.test.com";
+    }
+}
+
+describe("ApiClient", () => {
+    let client: TestApiClient;
+
+    beforeEach(() => {
+        client = new TestApiClient();
+        mockFetch.mockClear();
+    });
+
+    describe("buildUrl", () => {
+        test("should correctly build URL with normalized paths", () => {
+            const url = client.buildUrl("/test/path");
+            expect(url).toBe("https://api.test.com/test/path");
+        });
+
+        test("should handle paths with leading and trailing slashes", () => {
+            const url = client.buildUrl("/test/path/");
```

Comment:
Note: using `//test/path/` for example results in `https://api.test.com//test/path` which likely is expected behaviour. But I thought I would mention it here.